interface/src/app/mlmodel/resource.js

```diff
@@ -0,0 +1,9 @@
+angular.module('adage.mlmodel.resource', ['ngResource'])
+
+.factory('MLModel', ['$resource', function($resource) {
+  return $resource(
+    '/api/v0/mlmodel/'
+  );
+}])
```

Comment:
Either way will be fine with me. When I first AngularJS last year, I liked to have separate API services when possible. (You can find them in my old code). The reason why I am getting more conservative is mainly because of two reasons:
(1) Sometimes they are so simple, just returns a `$resource(url)`;
(2) Many services created this way were actually only used in the current module.

Originally I thought that creating a service may improve the frontend efficiency, because a service is always a singleton. But if you read $resource doc, you will see that `$resource(url)` actually returns a light-weight "class" type (which is a constructor function). All the weight-lifting part will be done later by the methods such as `get`, `post`, etc.

Anyway, my two cents.